pkg/field/struct_test.go

```
@@ -0,0 +1,32 @@
+		FieldsRequiredTogether(bf, ssf),
+	}
+
+	config := NewConfiguration(fields, constraints...)
+
+	data, err := json.Marshal(config)
+	if err != nil {
+		t.Fatalf("Failed to marshal Configuration: %v", err)
+	}
+
+	// fmt.Printf("JSON Output: \n\n%s\n\n", data)
+	expected := `{"Fields":[{"FieldName":"ss","FieldType":"StringField","Description":"Field 1","Rules":{},"DefaultValue":"default"},{"FieldName":"if","FieldType":"IntField","Description":"Field 2","Rules":{},"DefaultValue":42},{"FieldName":"bf","FieldType":"BoolField","Description":"Field 3","Rules":{}},{"FieldName":"ssf","FieldType":"StringSliceField","Description":"Field 4","Rules":{},"DefaultValue":["default"]}],"Constraints":[{"Kind":"MUTUALLY_EXCLUSIVE","Fields":["ss","if"]},{"Kind":"REQUIRED_TOGETHER","Fields":["bf","ssf"]}]}`
```

golangci-lint (1.62.2)

[error] line-length-limit: line is 539 characters, out of limit 200

(revive)